code/game/turfs/turf_changing.dm

```diff
@@ -39,30 +40,27 @@
 	var/old_lighting_overlay = lighting_overlay
 	var/old_corners = corners
 	var/old_ao_neighbors = ao_neighbors
+	var/old_permit_ao = permit_ao
 	var/old_above = above
 	var/old_prev_type = prev_type
-
-//	log_debug("Replacing [src.type] with [N]")
+	var/old_density = density
 
 	changing_turf = TRUE
 
-	if(connections) connections.erase_all()
+	if(connections) 
+		connections.erase_all()
 
 	overlays.Cut()
 	underlays.Cut()
 
-	// Run the Destroy() chain.
 	qdel(src)
 
 	var/old_opaque_counter = opaque_counter
-	var/turf/simulated/W = new N(src)
+	var/turf/W = new N(src)
```

Comment:
If you're changing the type of this var, why keep it at all?
As far as I can tell, it was just used to get access to procs on `/simulated`.

Reply:
Imo it's clearer than src manipulation.